saleor/order/utils.py

```diff
@@ -133,6 +133,8 @@ def recalculate_order_prices(order: Order, **kwargs):
     voucher_discount = kwargs.get("discount", zero_money(order.currency))
 
     # discount amount can't be greater than order total
+    if type(voucher_discount) == TaxedMoney:
```

Comment:
Wouldn't be better to call `isinstance`?

Reply:
As in the first `if` condition. If you're receiving here voucher_discount as a TaxedMoney, I would say that we should fix the place when this value is set to TaxedMoney, not by converting them to gross (which is also not ideal behaviour as .gross/.net should depend here on Saleor's setting `include_taxes_in_prices`)